src/qat/purr/compiler/builders.py

```diff
@@ -147,7 +157,9 @@ def merge_builder(self, other_builder: "InstructionBuilder"):
             )
             other_builder._fix_clashing_label_names(name_clashes, self.existing_names)
 
-        self.add(other_builder._instructions)
+        index = index or len(self._instructions)
+        self.insert(other_builder._instructions, index)
+        return index + len(other_builder._instructions)
```

Comment:
what does the return value mean and what's it intended for ?

Reply:
The index to insert the next instruction at.